x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.ts

```diff
@@ -92,8 +93,8 @@ export const CredentialsLogic = kea<CredentialsLogicType>({
     showCredentialsForm: (apiToken = { ...defaultApiToken }) => apiToken,
     hideCredentialsForm: false,
     resetCredentials: false,
-    initializeCredentialsData: true,
-    fetchCredentials: (page) => page,
+    onPaginate: (newPageIndex) => ({ newPageIndex }),
+    fetchCredentials: true,
```

Comment:
I removed the passed `page` arg from `fetchCredentials()` in favor of creating a new `onPaginate` action that handles updating the `meta` state. This accomplishes 2 things:

1. It treats `meta.page` as the local source of truth
2. It triggers a new fetch as a side-effect rather than manually (see above comment on the new `useEffect` code)